pkg/policies/flowcontrol/service/envoy/authz.go

```diff
@@ -139,48 +133,45 @@ func (h *Handler) Check(ctx context.Context, req *ext_authz.CheckRequest) (*ext_
 			ctrlPt = selectors.NewControlPoint(flowcontrolv1.ControlPointInfo_TYPE_EGRESS, "")
 		default:
 			// TODO(krdln) metrics
-			log.Sample(invalidTrafficDirectionSampler).
-				Warn().Str("traffic-direction", trafficDirectionHeader).Msg("invalid traffic-direction")
-			return nil, grpc_status.Error(grpc_codes.InvalidArgument, "invalid traffic-direction")
+			return nil, grpc.LoggedError(log.Sample(invalidTrafficDirectionSampler).Warn()).
+				Code(codes.InvalidArgument).Msg("invalid traffic-direction")
 		}
 	} else {
 		// TODO(krdln) metrics
-		log.Sample(missingTrafficDirectionSampler).
-			Warn().Msg("missing traffic-direction")
-		return nil, grpc_status.Error(grpc_codes.InvalidArgument, "missing traffic-direction")
+		return nil, grpc.LoggedError(log.Sample(missingTrafficDirectionSampler).Warn()).
+			Code(codes.InvalidArgument).Msg("missing traffic-direction")
 	}
 
-	var svcs []string
 	rpcPeer, peerExists := peer.FromContext(ctx)
-	if peerExists {
-		clientIP := strings.Split(rpcPeer.Addr.String(), ":")[0]
-		if h.entityCache != nil {
-			entity, err := h.entityCache.GetByIP(clientIP)
-			if err == nil {
-				svcs = entity.Services
-			}
-		}
+	if !peerExists {
+		return nil, grpc.Bug().Msg("cannot get peer info")
 	}
 
+	clientIP := strings.Split(rpcPeer.Addr.String(), ":")[0]
+	entity, err := h.entityCache.GetByIP(clientIP)
+	if err != nil {
+		return nil, grpc.LoggedError(log.Sample(unknownEntitySampler).Warn()).
+			Str("IP", clientIP).Code(codes.NotFound).Msg("unknown entity")
```

Comment:
Entity lookup failure should be treated fatal. Service discovery is disabled in sidecar based installation. The catch all service rule still applies.

If there is a genuine discovery issue, it can be discovered via Flow Analytics or via other mechanisms